### PR TITLE
TP-26790 Python 3 upgrade: django-pg-extensions upgrade

### PR DESCRIPTION
**Pull request by michalplichta-sc**
Python 3 upgrade: django-pg-extensions upgrade
https://scurri.tpondemand.com/entity/26790-python-3-upgrade-django-pg-extensions

As a Scurri developer, I would like to upgrade the django-pg-extensions library to support Python 2.7 and Python 3.4+.

The library can be found here:
https://github.com/Scurri/django-pg-extensions

Tests using `tox` must be implemented to ensure that the upgraded library is supported by both, Python 2.7 and 3.4+.

We must also bump the version so we can later update Scurri requirements.

**Acceptance Criteria**

* The library is upgraded
* Proper tests are implemented
* The version is bumped

**Goal**
Make the library Python 2 and 3 compatible

<details>
<summary>Coverage</summary>

**Python 2.7:**
```
---------- coverage: platform linux2, python 2.7.6-final-0 -----------
Name                                       Stmts   Miss  Cover   Missing
------------------------------------------------------------------------
djangopg/__init__.py                           0      0   100%
djangopg/copy.py                              59      2    97%   9-10
djangopg/fields.py                            65      0   100%
djangopg/models.py                             0      0   100%
djangopg/postgresql_psycopg2/__init__.py       2      0   100%
djangopg/postgresql_psycopg2/base.py          13      0   100%
djangopg/where.py                              9      0   100%
------------------------------------------------------------------------
TOTAL                                        148      2    99%
```

**Python 3.4:**
```
----------- coverage: platform linux, python 3.4.3-final-0 -----------
Name                                       Stmts   Miss  Cover   Missing
------------------------------------------------------------------------
djangopg/__init__.py                           0      0   100%
djangopg/copy.py                              59      1    98%   28
djangopg/fields.py                            65      0   100%
djangopg/models.py                             0      0   100%
djangopg/postgresql_psycopg2/__init__.py       2      0   100%
djangopg/postgresql_psycopg2/base.py          13      0   100%
djangopg/where.py                              9      0   100%
------------------------------------------------------------------------
TOTAL                                        148      1    99%
```
</details>

**Checks**
 - [x] Acceptance Criteria Satisfied
 - [x] Code changes are covered by test suite
 - [x] Pull request is properly formatted

### CODE REVIEW

File: djangopg/copy.py [hunk 1]
```diff
@@ -17,8 +23,9 @@ def _convert_to_csv_form(data):
     if data == '':
         return '""'
     # CSV needs to be encoded to UTF8
-    if isinstance(data, unicode):
-        return data.encode('UTF-8')
+    if isinstance(data, six.text_type):
+        if six.PY2:
```
Comment (brian-barba-hernandez) [inline djangopg/copy.py]: Join these 2 ifs together, please

File: djangopg/fields.py [hunk 2]
```diff
@@ -73,20 +72,20 @@ def db_type(self, connection):
         return 'citext'
 
     def to_python(self, value):
-        if isinstance(value, unicode) or value is None:
+        if isinstance(value, six.text_type) or value is None:
             return value
         if isinstance(value, str):
```
Comment (brian-barba-hernandez) [inline djangopg/fields.py]: This line may refer to binary type, on Python 3:
```
>>> value = str('123')
>>> if isinstance(value, str):
...     value.decode('UTF-8')
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AttributeError: 'str' object has no attribute 'decode'
```
Please, make sure the coverage of this library is 100%, this library is critical.

**Review by brian-barba-hernandez — CHANGES_REQUESTED** [review]

**Comment by michalplichta-sc** [comment]
### Coverage report for Python 3.4
```
Name                                       Stmts   Miss  Cover   Missing
------------------------------------------------------------------------
djangopg/__init__.py                           0      0   100%
djangopg/copy.py                              56     10    82%   27, 51-61
djangopg/fields.py                            46      4    91%   38-45
djangopg/models.py                             0      0   100%
djangopg/postgresql_psycopg2/__init__.py       2      2     0%   3-5
djangopg/postgresql_psycopg2/base.py          13     13     0%   3-27
djangopg/test_settings.py                      5      0   100%
djangopg/where.py                              9      0   100%
------------------------------------------------------------------------
TOTAL                                        131     29    78%
```

File: djangopg/copy.py [hunk 3]
```diff
@@ -45,7 +51,7 @@ def _send_csv_to_postgres(csv_text, conn, table_name, columns):
     fd = StringIO(csv_text)
     # Move the fp to the beginning of the string
     fd.seek(0)
-    columns = map(conn.ops.quote_name, columns)
+    columns = list(map(conn.ops.quote_name, columns))
```
Comment (brian-barba-hernandez) [inline djangopg/copy.py:59]: Please cover this change with tests for the `_send_csv_to_postgres` method.

**Review by brian-barba-hernandez — CHANGES_REQUESTED** [review]
Also, as I suggested in slack, please, don't add separated comments for the coverage, add then in the PR description at the top, easier to spot it.

**Review by brian-barba-hernandez — APPROVED** [review]

**Review by dmitrynedbaylo-sc — CHANGES_REQUESTED** [review]
Please read chat https://scurri.slack.com/archives/C02K3BV0266/p1653404899794079

**Comment by brian-barba-hernandez** [comment]
Closing as I'll be creating a new PR which will be cleaner

**Pull request closed without merging**